app/models/manageiq/providers/ansible_tower/shared/inventory/parser/automation_manager.rb

```diff
@@ -65,6 +67,12 @@ def configuration_script_sources
       inventory_object.scm_update_on_launch = project.scm_update_on_launch
       inventory_object.status = project.status
       inventory_object.last_updated_on = project.last_updated
+      inventory_object.last_update_error = nil
+
+      unless inventory_object.status == 'successful'
+        last_update = project.last_update
+        inventory_object.last_update_error = last_update.result_stdout.mb_chars.limit(ERROR_MAX_SIZE) if last_update
```

Comment:
Curious on why you chose `.mb_chars.limit` over just `.truncate`.  We use the latter, so it's preferable unless you have a specific reason.

Reply:
If you're trying to solve issues with result_stdout being non-UTF-8 (US-ASCII or ASCII-8BIT) and being afraid of chopping multibyte chars in half, you could also just do the following

```ruby
result_stdout.force_encoding("UTF-8").truncate(ERROR_MAX_SIZE)
```

(mb_chars assumes your incoming String is always UTF-8 anyway)

Reply:
Additionally, I think you need a spec for this truncation.  If you `stub_const(ManageIQ::Providers::AnsibleTower::Shared::Inventory::Parser::AutomationManager::ERROR_MAX_SIZE, 10)` or some really small number, then you can see if it truncates it to that amount.

Reply:
cc @tumido

Reply:
@Fryguy I've chose to use `.mb_chars.limit` over `.truncate` because we want to limit the data to a certain size not to a certain string length. [`.truncate` counts characters not bytes](https://apidock.com/rails/String/truncate). If we want to use `truncate` we shouldn't have a constant named `ERROR_MAX_SIZE` but rather `ERROR_MAX_LENGTH` and it shouldn't be in bytes...

```ruby
> "ááááááááááá".mb_chars.limit(8).bytesize
8
> "ááááááááááá".truncate(8).bytesize
13
```